src/ReferenceList.php

```diff
@@ -126,7 +126,7 @@ public function hasReference( Reference $reference ) {
 	 */
 	public function indexOf( Reference $reference ) {
 		foreach ( $this->references as $index => $ref ) {
-			if ( $ref->equals( $reference ) ) {
+			if ( $ref === $reference ) {
```

Comment:
As I already tried to explain: If it was possible to overlook a breaking change like this, with obviously no test, what else is untested and broken now? 1000 lines got deleted, 200 added (see https://github.com/wmde/WikibaseDataModel/compare/4.4.0...master).

Before doing any breaking change to the ReleaseList class I would have rewritten the whole test, adding dozens of small test cases for all public features of the class (including everything SPL related). Then it would be way, way more obvious what feature breaks and what not.

Now we have no good idea and no easy way to find out and be sure.